rollbar-java/build.gradle

```diff
@@ -12,12 +12,66 @@ buildscript {
 apply plugin: "nebula.integtest"
 
 dependencies {
-    api project(':rollbar-api')
+  api project(':rollbar-api')
 
-    api group: 'org.slf4j', name: 'slf4j-api', version: '1.7.25'
+  api group: 'org.slf4j', name: 'slf4j-api', version: '1.7.25'
 
-    compileOnly 'com.google.code.findbugs:jsr305:3.0.2'
+  compileOnly 'com.google.code.findbugs:jsr305:3.0.2'
 
-    integTestImplementation group: 'com.github.tomakehurst', name: 'wiremock', version: '2.27.0'
-    integTestImplementation group: 'com.google.code.gson', name: 'gson', version: '2.8.2'
+  integTestImplementation group: 'com.github.tomakehurst', name: 'wiremock', version: '2.27.0'
+  integTestImplementation group: 'com.google.code.gson', name: 'gson', version: '2.8.2'
+}
+
+/**
+ * This task will create a version property that is statically referenced when populating the
+ * `notifier` section of the payload. It helps when users shade and / or relocate the
+ * `rollbar-java` classes, since in those cases we no longer have access to our jar manifest.
+ * The task creates a Java class instead of a text resource, since dynamically loaded resources
+ * are not as reliable under relocation as a strongly typed bytecode reference to a compiled class.
+ */
+task createVersionClass() {
+  ext.set("OUTPUT_DIR", [buildDir.getAbsolutePath(), 'src', 'generated', 'main'].join(File.separator))
+
+  outputs.dir(ext.OUTPUT_DIR)
+
+  doLast {
+    def pkg = ["com", "rollbar", "notifier", "provider", "notifier"];
+
+    def pkgName = pkg.join(".");
+    def pkgPath = "${ext.OUTPUT_DIR}${File.separator}" +
+            (['src', 'generated'] + pkg).join(File.separator)
```

Comment:
Do we want this path? Seems that we have two times `src/generated` in the path, just double-checking.

Reply:
Yup, intellij even warned me about it I thought it was a false positive... :facepalm:

Reply:
Fixed in the latest push